plugins/http2_adapter/lib/src/http2_adapter.dart

```diff
@@ -196,11 +196,34 @@ class Http2Adapter implements HttpClientAdapter {
     // Handle redirection.
     if (needRedirect) {
       final url = responseHeaders.value('location');
+      if (url == null) {
```

Comment:
Do we have a RFC for the behavior? I can come out with another idea that we might hanfle this as a self-redirection. Someone else could thought the same thing.

Reply:
No, that just didn't make sense to me. Needs to be changed back.

Reply:
Fixed